File: Path Sum leetcode 112/Solution.java
class Node{
    int data;
    Node left;
    Node right;

    public Node(int data, Node left, Node right) {
        this.data = data;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public boolean hasSum(Node root, int currSum, int targetSum){
        if (root == null){
            return false;
        }
        currSum += root.data;

        if(root.left == null && root.right == null){
            return currSum == targetSum;
        }

        return hasSum(root.left, currSum, targetSum) || hasSum(root.right, currSum, targetSum);
    }
    public boolean hasPathSum(Node root, int targetSum) {
        return hasSum(root, 0, targetSum);
        
    }
}

File: Path Sum leetcode 112/Solution.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
        
class Solution:
    def hasPathSum(self, root: Node, targetSum: int)-> bool:
        
        def has_sum(root, currSum):
            if not root:
                return False
            
            currSum += root.data
            
            if not root.left and not root.right:
                return currSum == targetSum
            
            return has_sum(root.left, currSum) or has_sum(root.right, currSum)
        
        return has_sum(root, 0)